lib/HttpClient.js

```diff
@@ -223,6 +223,28 @@ function rawRequest(opts, cb) {
     });
     req.log = log;
 
+    var startRequestTimeout = function startRequestTimeout() {
+        if (opts.requestTimeout) {
+            requestTimer = setTimeout(function requestTimeout() {
+                requestTimer = null;
+
+                var err = new RequestTimeoutError(opts.requestTimeout);
+                dtrace._rstfy_probes['client-error'].fire(function () {
+                    return ([id, err.toString()]);
+                });
+
+                cb(err, req);
```

Comment:
Why do we call cb before `req.abort()`? Will we have the correct request state in the callback?

Reply:
The cb here is an internal cb that sets up a listener on the `result` event below. That listener is what triggers the user land callback, so in this case it looks like this is the correct flow. Great question though! Not immediately obvious that this is the case.